_posts/2020-11-06-shell-computing.markdown

```diff
@@ -0,0 +1,43 @@
+---
+layout: post
+title:  "Introduction to Shell Computing"
+date:   2020-11-08 12:54:04 -0700
+categories: shell computing
+---
+**What is a shell?**
+When you think of a shell what may first come to mind is the outer covering of an animal such as a crab or a turtle. However, when you leave the animal world and enter into the computer world there is another kind of shell.
+
+In the case of computer science, a shell is the tool a user can utilize in order to interact with the components of a computer. A shell is essentially a software program that is able to interpret commands from the user so the operating system can understand them and perform the various functions. It got the name shell because it is the outermost layer around the operating system. Shells can run on local systems but there is a way to make remote systems available as well making it possible to access remote servers. For example, our server at University of San Francisco, does not have a graphical interface so in this case we need to use the command line to access it in order to do our projects.   
+
+**Command-line interface vs graphical user interface**
```

Comment:
Nice explanation for these two types of interface. I would recommend to add a picture for both interface to help beginners virtualize the difference between CLI and GUI.